packages/react/src/components/Masthead/Masthead.js

```diff
@@ -5,34 +5,146 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import PropTypes from 'prop-types';
 import React from 'react';
+import PropTypes from 'prop-types';
 import MastheadSearch from './MastheadSearch';
+import { settings } from 'carbon-components';
+import HeaderContainer from 'carbon-components-react/lib/components/UIShell/HeaderContainer';
+import UserProfile20 from '@carbon/icons-react/lib/user--profile/20';
+import { ReactComponent as Logo } from '../Icon/svg/ibm-logo.svg';
+import {
+  Header,
+  HeaderMenuButton,
+  HeaderGlobalBar,
+  HeaderGlobalAction,
+  HeaderName,
+  HeaderNavigation,
+  HeaderMenu,
+  HeaderMenuItem,
+  HeaderSideNavItems,
+  SkipToContent,
+  SideNav,
+  SideNavItems,
+  SideNavLink,
+  SideNavMenu,
+  SideNavMenuItem,
+} from 'carbon-components-react/lib/components/UIShell';
+import '@ibmdotcom/styles/scss/components/masthead/_masthead.scss';
+
+const { prefix } = settings;
 
 /**
- * Test MastHead component
+ * MastHead component
  *
+ * @typedef {object} navigation Object containing navigation elements
  * @param {string} type Type of masthead
  * @returns {*} Masthead component
- * @class
  */
-const Masthead = ({ type }) => {
-  if (type === 'branded') {
-    return (
-      <div className="masthead">
-        <MastheadSearch />
-      </div>
-    );
-  } else {
-    return (
-      <div className="masthead">
-        <MastheadSearch />
-      </div>
-    );
-  }
+const Masthead = ({ navigation, type }) => {
+  return (
+    <HeaderContainer
```

Comment:
@kennylam can you add in a stable selector wrapper here? And probably any other stable selector tags in the header that probably make sense:
https://github.com/carbon-design-system/ibm-dotcom-library/blob/master/docs/stable-selectors.md